crates/bevy_scene/src/scene_filter.rs

```diff
@@ -154,6 +155,9 @@ impl SceneFilter {
     ///
     /// [`Unset`]: SceneFilter::Unset
     pub fn is_allowed_by_id(&self, type_id: TypeId) -> bool {
+        if World::is_internal_type(type_id) {
+            return false;
+        }
```

Comment:
`SceneFilter` is meant to be somewhat general purpose and allow for any type to be filtered. As such, I think this code should remain the same and this sort of hardcoding done in the actual scene logic like you already have.

Note that eventually we do want to have constructors for `SceneFilter` that come with sensible defaults, but that's likely outside the scope of this PR as it has a few decisions/problems that need working out.